packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -69,6 +83,9 @@ function useMediaUploadQueue() {
   const { uploadVideoPoster } = useUploadVideoFrame({
     updateMediaElement: noop,
   });
+
+  const isMounted = useRef(false);
```

Comment:
Why is this is needed?

Reply:
To prevent state updates after the component has already unmounted.

Mostly relevant for the tests to clean things up properly.

Otherwise you get errors, example:

1. Init stuff
2. Run async process
3. Hook is unmounted
4. Async process finishes, tries to update state
5. Error because component has unmounted already

I'm sure you've seen this message in the wild before at some point:

```
Warning: Can 't perform a React state update on an unmounted component. This is a no - op, but it indicates a memory leak in your application
```